src/Agent.Worker/Release/ReleaseJobExtension.cs

```diff
@@ -0,0 +1,258 @@
+                            }
+                            else
+                            {
+                                var buildArtifactProvider = HostContext.GetService<IArtifactProvider>();
+                                await buildArtifactProvider.Download(executionContext, artifactDefinition, downloadFolderPath);
+                            }
+                        });
+
+                executionContext.Output(StringUtil.Loc("RMArtifactDownloadFinished", agentArtifactDefinition.Alias));
+            }
+
+            executionContext.Output(StringUtil.Loc("RMArtifactsDownloadFinished"));
+        }
+
+        private static string GetLocalDownloadFolderPath(AgentArtifactDefinition artifactDefinition, string artifactsWorkingFolder)
+        {
+            return  Path.GetFullPath(Path.Combine(artifactsWorkingFolder, artifactDefinition.Alias ?? string.Empty));
+        }
+
+        private void CleanUpArtifactsFolder(IExecutionContext executionContext, string artifactsWorkingFolder)
+        {
+            Trace.Entering();
+            executionContext.Output(StringUtil.Loc("RMCleaningArtifactsDirectory", artifactsWorkingFolder));
+            try
+            {
+                var releaseFileSystemManager = HostContext.GetService<IReleaseFileSystemManager>();
+                releaseFileSystemManager.DeleteDirectory(artifactsWorkingFolder);
+            }
+            catch (Exception ex)
+            {
+                Trace.Error(ex);
+                // Do not throw here
+            }
+            finally
+            {
+                executionContext.Output(StringUtil.Loc("RMCleanedUpArtifactsDirectory", artifactsWorkingFolder));
+            }
+        }
+
+        private void InitializeAgent(IExecutionContext executionContext, out bool skipArtifactsDownload, out Guid teamProjectId, out string artifactsWorkingFolder, out int releaseId)
+        {
+            Trace.Entering();
+
+            releaseId = executionContext.Variables.GetInt(Constants.Variables.Release.ReleaseId) ?? 0;
+            teamProjectId = executionContext.Variables.GetGuid(Constants.Variables.System.TeamProjectId) ?? Guid.Empty;
+            skipArtifactsDownload = executionContext.Variables.GetBoolean(Constants.Variables.Release.SkipArtifactsDownload) ?? false;
+            string releaseDefinitionName = executionContext.Variables.Get(Constants.Variables.Release.ReleaseDefinitionName);
+
+            // TODO: Should we also write to log in executionContext.Output methods? so that we don't have to repeat writing into logs?
+            // Log these values here to debug scenarios where downloading the artifact fails.
+            executionContext.Output(
+                $"ReleaseId={releaseId}, TeamProjectId={teamProjectId}, ReleaseDefinitionName={releaseDefinitionName}");
+
+            var configStore = HostContext.GetService<IConfigurationStore>();
+            var shortHash = ReleaseFolderHelper.CreateShortHash(
```

Comment:
Does this get cleaned up? We ended up ditching the short hash for build and switched to an incrementing integer. IIRC one of the driving factors was that our short hash was too short (chance for collision). Something to think about.